cmd/pinniped-proxy/src/https.rs

```diff
@@ -73,11 +80,56 @@ pub fn cert_for_cert_data(cert_data: Vec<u8>) -> Result<Certificate> {
     }
 }
 
+/// rewrite_request directs the specified request to the targeted backend api
+/// server.
+pub fn rewrite_request(mut req: Request<hyper::Body>, k8s_api_server_url: String) -> Result<Request<hyper::Body>> {
+    // Update the request URI.
+    let uri_string = format!(
+        "{}{}",
+        k8s_api_server_url,
+        req.uri().path_and_query().map(|x| x.as_str()).unwrap_or("")
+    );
+    let uri: hyper::Uri = uri_string.parse()?;
+    *req.uri_mut() = uri.clone();
+
+    // Update the host header.
```

Comment:
I'm not sure, but, since we are proxing here, should we add the [`X-Forwarded-xxxx` ](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Forwarded) headers also?

Reply:
I had done this initially, then removed as in this case, there is no info being lost during the proxying since the request comes with the target destination in the headers (ie. it's not a reverse proxy hiding backend services, but an authenticating proxy for known backend services). I'll have another think about it, let me know if you've more thoughts.